src/test/scala/org/broadinstitute/dsde/workbench/sam/service/TosServiceSpec.scala

```diff
@@ -73,4 +67,38 @@ class TosServiceSpec extends AnyFlatSpec with TestSupport with BeforeAndAfterAll
     tosServiceEnabledV2.acceptTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(true)
     tosServiceEnabledV2.getTosStatus(defaultUser.id, samRequestContext).unsafeRunSync() shouldBe Option(true)
   }
+
+  "TosService.isTermsOfServiceStatusAcceptable" should "allow all requests to the API if TOS is disabled" in {
+    assume(databaseEnabled, databaseEnabledClue)
+    dirDAO.createUser(defaultUser, samRequestContext).unsafeRunSync()
+    tosServiceDisabled.isTermsOfServiceStatusAcceptable(defaultUser) should be(true)
+  }
+
+  it should "not allow users who have never accepted a ToS version to use the API, even with a grace period enabled" in {
+    assume(databaseEnabled, databaseEnabledClue)
+    dirDAO.createUser(defaultUser, samRequestContext).unsafeRunSync()
+    tosServiceGracePeriodEnabled.isTermsOfServiceStatusAcceptable(defaultUser) should be(false)
+  }
+
+  it should "allow users that have accepted a previous ToS version to use the API when the grace period is enabled" in {
+    assume(databaseEnabled, databaseEnabledClue)
+    dirDAO.createUser(defaultUser, samRequestContext).unsafeRunSync()
+    dirDAO.acceptTermsOfService(defaultUser.id, "1", samRequestContext).unsafeRunSync()
+    val userAcceptedPreviousVersion = dirDAO.loadUser(defaultUser.id, samRequestContext).unsafeRunSync().orNull
+    tosServiceGracePeriodEnabled.isTermsOfServiceStatusAcceptable(userAcceptedPreviousVersion) should be(true)
+  }
+
+  it should "not allow users that have accepted a previous ToS to use the API without a grace period" in {
+    assume(databaseEnabled, databaseEnabledClue)
+    dirDAO.createUser(defaultUser, samRequestContext).unsafeRunSync()
+    dirDAO.acceptTermsOfService(defaultUser.id, "1", samRequestContext).unsafeRunSync()
+    val userAcceptedPreviousVersion = dirDAO.loadUser(defaultUser.id, samRequestContext).unsafeRunSync().orNull
+    tosServiceEnabledV2.isTermsOfServiceStatusAcceptable(userAcceptedPreviousVersion) should be(false)
```

Comment:
Might want to try being more explicit with these tests overall.  I know there are default settings coming in, and that's not bad, but these tests would be even better if you can set up the ToS config.  For example, in all of these tests, I don't know if ToS is enabled or Disabled or what the starting state of the grace period is, I need to go look that up in the config.  I think to truly test all the scenarios here, we need to be able to set the config into all sorts of different states.  I don't want to necessarily drag this fix out too long, but I think just taking a second look at this to see if you can be more explicit within each test, these could be better and we can do a better job making sure we have all the possible scenarios covered.

Reply:
I've gone through the new (and existing) tests and made it more explicit about the state of `TosService`